lib/assert.js

```diff
@@ -341,6 +359,233 @@ assert.notStrictEqual = function notStrictEqual(actual, expected, message) {
   }
 };
 
+function isPlainObject(obj) {
+  if (typeof obj !== 'object' || obj === null) {
+    return false;
+  }
+  const proto = ObjectGetPrototypeOf(obj);
+  return proto === ObjectPrototype || proto === null || ObjectPrototypeToString(obj) === '[object Object]';
+}
+
+function assertIncludes(actual, expected, loose) {
+  if (expected.length > actual.length) {
+    return false;
+  }
+  if (typeof actual === 'string') {
+    return StringPrototypeIncludes(actual, expected);
+  }
+
+  return ArrayPrototypeEvery(expected, (item) => ArrayPrototypeSome(actual, (actualItem) => {
+    // eslint-disable-next-line eqeqeq
+    return loose ? actualItem == item : actualItem === item;
+  }));
```

Comment:
IMO we should just be using `Array.prototype.includes`, this really should only belong in `partialDeepEqual`. Having `assert.includes` makes perfect sense, having `assert.includesStrict` less so (or it should be an alias maybe).

Reply:
I would be kinda agreeing with you if most of the methods didn't have its own `strict` variant. If we would do what you suggested, we would have `includes` which is not doing type coercion, going agains the pattern we have for pretty much the whole `assert` module

Reply:
Another path would be to remove `includes` from this PR so `partialDeepStrictEqual` can land, and we can discuss the `includes` (and possibly `includesStrict`) in another PR, wdyt?

Reply:
@aduh95 it kinda bothers me to remove functionality considering it had been implemented already 😄 But  I understand your concerns. I would ping people to give us more feedback instead tho. We could give us a couple of days?

Reply:
Sure I was suggesting that in case you'd like the `partialDeepStrictEqual` to land without waiting since it has consensus, but it's certainly fine to wait.

Reply:
what about a small input from @danielbayley ? He was the one introducing the idea of adding `includes` and `includesStrict` into the PR, getting a bunch of upvotes too

Reply:
I tried to look for that proposal, is it https://github.com/nodejs/node/issues/50399#issuecomment-2296531960? In this comment, Daniel seems to agree with me that `assert.includes` should mirror the built-in JS methods, like I'm suggesting. Or am I missing something?

Reply:
@aduh95 I might have misunderstood the previous discussion then!

1. removed `includesStrict`
2. adapted `includes` to be mirroring the native `includes`
3. adapted the documentation

thanks again!